pytorch_lightning/utilities/fetching.py

```diff
@@ -35,34 +38,60 @@
 class AbstractDataFetcher(ABC):
 
     """
-    This class is used to control batch fetching flow.
+    This based class should be used to implement a fault tolerant ``DataFetcher``.
+    It is required to override the ``fetching_function`` with fetching logic.
+    Example::
+        class SimpleDataFetcher(AbstractDataFetcher):
+            def fetching_function(self):
+                while True:
+                    try:
+                        yield next(self.dataloader_iter), False
+                    except StopIteration:
+                        yield None, True
     """
 
     @abstractmethod
     def fetching_function(self) -> Generator:
-        pass
+        """Override with your own fetching logic."""
 
     def __init__(
```

Comment:
why is init below the fetching function 😄

Reply:
I actually think it is good that way, since the fetching function is the abstract one to be overwritten

Reply:
Because it is a hook and should be placed at the top for better readability.